src/task.c

```diff
@@ -744,25 +755,30 @@ task_perform_one(task_system *ts)
 
 /*
  * Validate that the task system configuration is basically supportable.
- * This check protects us from configuring some bizillion #s of background
- * threads, which somehow sneak through other config parsing checks.
- * We also enforce other, somewhat arbitrary, limits to leave some headroom
- * for user-threads to exeecute.
  */
 static platform_status
 task_config_valid(uint8 num_bg_threads[NUM_TASK_TYPES])
 {
    uint64 normal_bg_threads   = num_bg_threads[TASK_TYPE_NORMAL];
    uint64 memtable_bg_threads = num_bg_threads[TASK_TYPE_MEMTABLE];
 
-   if ((normal_bg_threads + memtable_bg_threads) > (MAX_THREADS / 2)) {
-      platform_error_log("Background thread configuration is not supported."
-                         " Total number of background threads configured"
-                         ", normal-bg-threads=%lu, memtable-bg-threads=%lu,"
-                         " must be <= %d\n",
+   if ((normal_bg_threads == 0) != (memtable_bg_threads == 0)) {
+      platform_error_log("Both configuration parameters for background "
+                         "threads, normal_bg_threads (%lu) "
+                         "and memtable_bg_threads (%lu) "
+                         "must be zero or be non-zero.\n",
+                         normal_bg_threads,
+                         memtable_bg_threads);
+      return STATUS_BAD_PARAM;
+   }
+
+   if ((normal_bg_threads + memtable_bg_threads) >= MAX_THREADS) {
+      platform_error_log("Total number of background threads configured"
```

Comment:
Reworked the checks on 'bad' configuration as discussed in the review.

Reply:
I think it's `>=` as the main thread uses slot 0. So bg-threads together can only come up to 63.